tests/test_cms_menus.py

```diff
@@ -404,6 +404,519 @@ def test_menu_without_softroots(self):
 
         self.assertTreeQuality(hard_root, mock_tree, 'level', 'title')
 
+    def test_menu_with_node_hidden(self):
+        """
+        Tree in fixture :
+               root
+                   aaa
+                       aaa1( Hide_node = True) hidden node
+                           ccc
+                               ddd
+                       aaa2
+                   bbb
+        tag: show_menu 0 100 0 100
+        expected result 1:
+               0:root
+                  1:aaa
+                     5:aaa2
+                  6:bbb
+        Note: If node has hide_node True , node and its descendants are removed from tree
+        """
+        root_pagecontent = factories.PageContentWithVersionFactory(
```

Comment:
It would be cleaner to move these commonly used page content variables into a setUp method.